### PR TITLE
removed properties from vm.header and block, all tests passed

### PR DESCRIPTION
**Pull request by dherykw**
### What was wrong?
class vm using properties to access and retreive values of block and header

### How was it fixed?
changing from properties to getter and setter methods in files where were requiered.

[//]: # (For important changes, please add a new entry to the running release notes PR)
[//]: # (You can find the current one using: https://github.com/ethereum/py-evm/pulls?q=is%3Aopen+is%3Apr+label%3A%22Release+Notes%22 )

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: eth/vm/base.py [hunk 1]
```diff
@@ -567,37 +562,41 @@ def import_block(self, block: BaseBlock) -> BaseBlock:
...
-                coinbase=block.header.coinbase,
-                gas_limit=block.header.gas_limit,
-                timestamp=block.header.timestamp,
-                extra_data=block.header.extra_data,
-                mix_hash=block.header.mix_hash,
-                nonce=block.header.nonce,
-                uncles_hash=keccak(rlp.encode(block.uncles)),
-            ),
-            uncles=block.uncles,
+        self.set_block(
```
Comment (pipermerriam) [inline eth/vm/base.py]: I'm inclined to advocate for removing this method in favor of directly setting via `self._block = ...`

The only place that `set_block` appears to be used is from the tests in `tests/json-fixtures/test_virtual_machine.py` which I'm fine with that accessing the private `vm._block` property for now.

File: eth/vm/base.py [hunk 2]
```diff
@@ -640,7 +639,7 @@ def set_block_transactions(self,
     #
     def _assign_block_rewards(self, block: BaseBlock) -> None:
         block_reward = self.get_block_reward() + (
-            len(block.uncles) * self.get_nephew_reward()
+                len(block.uncles) * self.get_nephew_reward()
```
Comment (pipermerriam) [inline eth/vm/base.py]: This indentation change looks like it should be reverted.

**Comment by dherykw** [comment]
any help?

**Comment by pipermerriam** [comment]
There are 4 linting errors you can see here: https://circleci.com/gh/ethereum/py-evm/132998

You should be able to see these locally if you run `tox -e py36-lint` which will run the full linter.  Looks like just indentation changes that need to be fixed.

**Comment by dherykw** [comment]
It seems that lint has been successfully fixed :) sorry for the inconvenience.

File: eth/vm/base.py (hunk 1 above)
Comment (dherykw) [reply eth/vm/base.py]: this is solved in the 9bd96bb commit

File: eth/vm/base.py (hunk 2 above)
Comment (dherykw) [reply eth/vm/base.py]: this was solved in the 9bd96bb commit

**Comment by pipermerriam** [comment]
Thank you for the contribution!

**Comment by dherykw** [comment]
My pleasure. I am looking forward continue collaborating. :-)

**Pull request merged**